config/main.py

```diff
@@ -2064,6 +2064,546 @@ def delete_snmptrap_server(ctx, ver):
     cmd="systemctl restart snmp"
     os.system (cmd)
 
+
+
+#
+# 'snmp' group ('config snmp ...')
+#
+@config.group(cls=clicommon.AbbreviationGroup, name='snmp')
+@clicommon.pass_db
+def snmp(db):
+    """SNMP configuration tasks"""
+
+
+@snmp.group(cls=clicommon.AbbreviationGroup)
+@clicommon.pass_db
+def community(db):
+    pass
+
+
+def is_valid_community_type(commstr_type):
+    commstr_types = ['RO', 'RW']
+    if commstr_type not in commstr_types:
+        click.echo("Invalid community type.  Must be either RO or RW")
+        return False
+    return True
+
+
+def is_valid_user_type(user_type):
+    convert_user_type = {'noauthnopriv': 'noAuthNoPriv', 'authnopriv': 'AuthNoPriv', 'priv': 'Priv'}
+    user_types = ['noauthnopriv', 'authnopriv', 'priv']
+    if user_type not in user_types:
```

Comment:
>user_types [](http://example.com/codeflow?start=24&length=10)

You can directly use `convert_user_type`. No need to define `user_types`.

Reply:
fixed.

Reply:
To further simplify, you can remove `.keys()`